some compile warning with MKLDNN related codes, etc

There are some compile warning with MKLDNN, the full log is https://paddleci.ngrok.io/viewLog.html?buildId=19199&buildTypeId=Paddle_PrCi&tab=buildLog:
1.  https://github.com/01org/mkl-dnn/issues/157
```
[09:53:57]W:	 [Step 1/1] /paddle/build/third_party/mkldnn/src/extern_mkldnn/src/cpu/jit_avx512_common_conv_kernel.cpp: In lambda function:
[09:53:57]W:	 [Step 1/1] /paddle/build/third_party/mkldnn/src/extern_mkldnn/src/cpu/jit_avx512_common_conv_kernel.cpp:2666:42: warning: assuming signed overflow does not occur when assuming that (X + c) < X is always false [-Wstrict-overflow]
[09:53:57]W:	 [Step 1/1]      auto emit_fma_block = [&](int kh_step) {
[09:53:57]W:	 [Step 1/1]                           
```
2. Trainer.cpp
```
[09:55:41]W:	 [Step 1/1] /paddle/paddle/trainer/Trainer.cpp:141:5:   required from here
[09:55:41]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:717:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[09:55:41]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_EQ, ==)  // Compilation error with CHECK_EQ(NULL, x)?
[09:55:41]W:	 [Step 1/1]                                 ^
[09:55:41]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:148:53: note: in definition of macro 'GOOGLE_PREDICT_TRUE'
[09:55:41]W:	 [Step 1/1]  #define GOOGLE_PREDICT_TRUE(x) (__builtin_expect(!!(x), 1))
[09:55:41]W:	 [Step 1/1]                                                      ^
[09:55:41]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:717:1: note: in expansion of macro 'DEFINE_CHECK_OP_IMPL'
[09:55:41]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_EQ, ==)  // Compilation error with CHECK_EQ(NULL, x)?
[09:55:41]W:	 [Step 1/1]  ^
```
3. beam_search_op.cc
```
[09:57:28]W:	 [Step 1/1] /paddle/paddle/operators/beam_search_op.cc: In member function 'bool paddle::operators::BeamSearch::NextItemSet(std::vector<paddle::operators::BeamSearch::Item>*)':
[09:57:28]W:	 [Step 1/1] /paddle/paddle/operators/beam_search_op.cc:142:23: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[09:57:28]W:	 [Step 1/1]      for (int d = 0; d < instance_dim; d++) {
[09:57:28]W:	 [Step 1/1]                        ^
```
4. test_MKLDNN.cpp
```
[10:13:00]W:	 [Step 1/1] /paddle/paddle/gserver/tests/test_MKLDNN.cpp:318:3:   required from here
[10:13:00]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:721:32: warning: comparison between signed and unsigned integer expressions [-Wsign-compare]
[10:13:00]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_GE, >=)
[10:13:00]W:	 [Step 1/1]                                 ^
[10:13:00]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:148:53: note: in definition of macro 'GOOGLE_PREDICT_TRUE'
[10:13:00]W:	 [Step 1/1]  #define GOOGLE_PREDICT_TRUE(x) (__builtin_expect(!!(x), 1))
[10:13:00]W:	 [Step 1/1]                                                      ^
[10:13:00]W:	 [Step 1/1] /paddle/build/third_party/install/glog/include/glog/logging.h:721:1: note: in expansion of macro 'DEFINE_CHECK_OP_IMPL'
[10:13:00]W:	 [Step 1/1]  DEFINE_CHECK_OP_IMPL(Check_GE, >=)
[10:13:00]W:	 [Step 1/1]  ^
```

--- a/paddle/gserver/layers/MKLDNNLayer.h
+++ b/paddle/gserver/layers/MKLDNNLayer.h
@@ -36,7 +36,7 @@ class MKLDNNLayer : public Layer {
 protected:
   // batch size
   int bs_;
-  // they sizes are always from the first input layer
+  // their sizes are always from the first input layer
   // input image channel, height and width
   int ic_, ih_, iw_;
   // output image channel, height and width

--- a/paddle/gserver/tests/test_MKLDNN.cpp
+++ b/paddle/gserver/tests/test_MKLDNN.cpp
@@ -315,7 +315,7 @@ TEST(MKLDNNLayer, AddtoLayer) {
 
 static void getMKLDNNConcatConfig(TestConfig& cfg,
                                   const std::vector<testImageDesc>& inputs) {
-  CHECK_GE(inputs.size(), 2) << "at least two inputs";
+  CHECK_GE(inputs.size(), 2UL) << "at least two inputs";
   int oc = inputs[0].ic;
   for (size_t i = 1; i < inputs.size(); ++i) {
     CHECK_EQ(inputs[i].bs, inputs[0].bs);

--- a/paddle/operators/beam_search_op.cc
+++ b/paddle/operators/beam_search_op.cc
@@ -139,7 +139,7 @@ bool BeamSearch::NextItemSet(std::vector<BeamSearch::Item> *items) {
   items->reserve(framework::product(ids.dims()));
   for (size_t offset = abs_lod[lod_level_][sent_offset_];
        offset < abs_lod[lod_level_][sent_offset_ + 1]; offset++) {
-    for (int d = 0; d < instance_dim; d++) {
+    for (size_t d = 0; d < instance_dim; d++) {
       const size_t dim_offset = offset * instance_dim + d;
       items->emplace_back(offset, ids_data[dim_offset],
                           scores_data[dim_offset]);

--- a/paddle/trainer/Trainer.cpp
+++ b/paddle/trainer/Trainer.cpp
@@ -138,7 +138,7 @@ void Trainer::init(const std::shared_ptr<TrainerConfigHelper>& config,
   }
 
   if (FLAGS_use_mkldnn) {
-    CHECK_EQ(FLAGS_trainer_count, 1UL) << "MKLDNN only need 1 trainer";
+    CHECK_EQ(FLAGS_trainer_count, 1) << "MKLDNN only need 1 trainer";
   }
 
   if (testing) {

